lib/MusicBrainz/Server/Report/SeriesContainingDuplicates.pm

```diff
@@ -0,0 +1,95 @@
+package MusicBrainz::Server::Report::SeriesContainingDuplicates;
+use Moose;
+
+use MusicBrainz::Server::Constants qw( %PART_OF_SERIES );
+use MusicBrainz::Server::Data::Utils qw( type_to_model );
+use MusicBrainz::Server::Entity::Util::JSON qw( to_json_object );
+
+with 'MusicBrainz::Server::Report::SeriesReport',
+     'MusicBrainz::Server::Report::FilterForEditor::SeriesID';
+
+around inflate_rows => sub {
+    my $orig = shift;
+    my $self = shift;
+
+    my $items = $self->$orig(@_);
+
+    for my $result (@$items) {
+        my $series = $result->{series};
+        my $series_type = $self->c->model('SeriesType')->get_by_id($series->{typeID});
+        my $entity_type = $series_type->item_entity_type;
+        my $model = $self->c->model(type_to_model($entity_type));
+        my $entity = $model->get_by_id($result->{entity_id});
+        $result->{entity} = to_json_object($entity);
+    }
+
+    return $items;
+};
+
+sub query {
+    my $self = shift;
+
+    my @subqueries;
+
+    for my $entity_type (keys %PART_OF_SERIES) {
+        my $part_of_series_gid = $PART_OF_SERIES{$entity_type};
+        my ($series_table_data) = $self->c->model('Relationship')->generate_table_list(
+            'series',
+            $entity_type,
+        );
+        my ($table_name, $series_column, $other_column) = @$series_table_data;
```

Comment:
I see you're calling `generate_table_list` in a loop because you need the target type, but I think it'd be nicer if we just changed `generate_table_list` to return the target type in its lists. Then you could just do:

```Perl
my @series_l_tables = $self->c->model('Relationship')->generate_table_list(
    'series',
    keys %PART_OF_SERIES,
);

for my $series_l_table (@series_l_tables) {
    my ($table_name, $series_column, $other_column, $target_type) = @$series_l_table;
    my $part_of_series_gid = $PART_OF_SERIES{$target_type};
    # ...
}
```

Reply:
Sure, I guess there's no big reason not to make this change, since other calls for this already take only parts of the result anyway. Made the change.